server/commands/list_objects.go

```diff
@@ -69,7 +103,36 @@ func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjec
 			user:        req.User,
 			ctxTuples:   req.ContextualTuples,
 		}, resultsChan, errChan, resolvedChan)
-	}()
+	}
+
+	// ConnectedObjects currently only supports models that do not include intersection and exclusion,
+	// and the model must include type info for ConnectedObjects to work.
+	if !typesystem.RewriteContainsIntersection(relation) &&
+		!typesystem.RewriteContainsExclusion(relation) &&
```

Comment:
What if I have a model like this

```
type document
    relations
         define viewer as self
         define restricted as self
         define editor as viewer but not restricted  
         define admin as editor
````

and then I call `ListObjects(user=maria, relation=admin, type: document)`

There is no exclusion in `admin` but it is defined as `as editor`, and `editor` does have exclusion. Will it work then?

Reply:
That case should default to the old behavior (concurrent checks). Let me add a test for that to make sure that the `typesystem` package's `RewriteContainsIntersection` and `RewriteContainsExclusion` methods behave this way. If they don't then we'll need to fix that behavior.

The idea is that if intersection or exclusion is involved whatsoever in the evaluation, then we should fall back to prior behavior.

Reply:
Yeah, this is an issue. These methods only operate directly on the relation's rewrite itself. It doesn't follow other relations across types. Good catch! Will fix tomorrow and make sure tests cover those cases.

Reply:
@miparnisari I have addressed this behavior now in the most recent changes. Would you mind taking a thorough second look and tell me what you think 👍 ?